### PR TITLE
Add missing api and some structures.

### PR DESCRIPTION
**Pull request by kekeimiku**
#22

### CODE REVIEW

**Comment by kekeimiku** [comment]
@JohnTitor Hi!

File: examples/dump_process_registers.rs [hunk 1]
```diff
@@ -43,7 +43,7 @@ fn read_int() -> Result<::libc::c_int, ()> {
             return Err(());
         }
     }
-    return Ok(value);
+    Ok(value)
```
Comment (JohnTitor) [inline examples/dump_process_registers.rs:46]: This should be unrelated to the API addiction, could you drop?

File: examples/dump_process_registers.rs [hunk 2]
```diff
@@ -81,7 +81,7 @@ fn main() {
         if kret != KERN_SUCCESS {
             println!("Did not succeed in getting task for pid {}", pid);
             println!("kern_return_t error {}", kret);
-            println!("");
+            println!();
```
Comment (JohnTitor) [inline examples/dump_process_registers.rs:84]: Ditto

File: src/libproc.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+//! This module roughly corresponds to `libproc.h`.
```
Comment (JohnTitor) [inline src/libproc.rs:1]: I'm not familiar with this, how does this relate to mach? It'd be great if it's documented.

**Pull request closed without merging**